### DISCUSSION TITLE
Plex connection error: Max retries exceeded with plex.direct.url - PTR uses old IP-Address

### DISCUSSION THREAD

**Discussion by gitttt54** [Taxel/PlexTraktSync#775] [Support] [answered]
Hello,

I installed a PlexTraktSync docker container and set it up working 2 weeks ago. I worked great and I did'nt run the container again until today.

Running the container today (starting the container via docker-tab in unraid, go to console, type `plextraktsync` or `python3 -m plextraktsync`) stops the container after a while, leaving this in the logfile:
```
2022-02-02 20:56:20,790 INFO[PlexTraktSync]:PlexTraktSync [0.18.15]
2022-02-02 20:56:39,935 ERROR[PlexTraktSync]:Plex connection error: HTTPSConnectionPool(host='92-31-59-112.7aec5c30ac8047b7988208159f4bd190.plex.direct', port=32241): Max retries exceeded with url: / (Caused by ConnectTimeoutError(<urllib3.connection.HTTPSConnection object at 0x14a09c24db40>, 'Connection to 92-31-59-112.7aec5c30ac8047b7988208159f4bd190.plex.direct timed out. (connect timeout=30)')), fallback url http://172.17.0.1:32400 didn't respond either.
```

If I'd be guessing, the ipaddress/URL shown in the log is from before 2 weeks and now -obviously, because it's a DSL connection I'm using- has changed, so PTS can not access the plex server. 
Port 32241 is extern port, which is forwarded on my router to the intern port 32400 on the host which is running plex. so that setup should be ok.

What I don't see, is why PTS is trying to connect to an old ip-address instead of ... well... making it right ;-) I have totally forgotten what I did when I installed and configured the container, but as far as I can see in the config and remember, there was nothing to set up with url/ip-addesses.

Question: Is this normal behaviour and I must configure the container freshly agian every time I start it after some time?
or what are my next steps to have the container running fine ?

Thank you

**Comment by glensc** [comment]
use plex-login command to authenticate again. or edit .env file and fill lan address:port

you can abort plex login command with ctrl+c if you just want to obtain connection addresses and then fill address manually to .env

if docs are missing in readme: be first one to submit pull request to improve docs!

**Comment by gitttt54** [comment]
Thank you, 
I have tried to enter the BASEURL in .env file, but that did not work (same log error as above).
So I just removed all the contents of the config dir and started from scratch. 
That worked fine and the sync started.

I aborted the sync, 
replaced the BASEURL in the .env file:
from 
PLEX_BASEURL=https://93-239-22-53.7aec5c30ac8456549882083453bd190.plex.direct:32241
to
PLEX_BASEURL=https://192.168.178.241:32400 (copied from my browser)
and restarted the container to be sure, the changes would affect.

Unfortunately,  I get the same error:

log:
_2022-02-20 17:09:25,340 INFO[PlexTraktSync]:PlexTraktSync [0.18.20]
2022-02-20 17:09:28,438 ERROR[PlexTraktSync]:Plex connection error: HTTPSConnectionPool(host='192.168.178.241', port=32400): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x1499d274e380>: Failed to establish a new connection: [Errno 113] Host is unreachable')), fallback url http://172.17.0.1:32400 didn't respond either._

IF I understood you advise correct, I need to connect PTS to my new replaced local URL via plex auth (but I'm not sure, I did).

So I commented out the PLEX_TOKEN line in the .env file and started plextraktsync again.
It then asked for my plex account and after I entered them, the sync started.

Unfortunately, the .env file now has again a *.plex.direct URL and not my local URL.

I guess, I did something completely wrong, but I can not see what (and how to do it right).

Anyone any ideas how I can use my local URL (https://192.168.178.241:32400) to sync ?

Thank you very much.

**Reply by glensc** [reply]
please edit your post to use markdown code fence, so github won't render the URLs.

if you enter 192.168 url directly, it must be `http://` not `https://` url. can't see what you entered because you didn't put your line to code block.

also, plex-login should print 192-168 URLs with plex.direct as well, copy that URL manually

**Comment by gitttt54** [comment]
I'm sorry, I have no idea what markdown code fence is and how to use it, but I edited my post anyways.

The great hint
>also, plex-login should print 192-168 URLs with plex.direct as well, copy that URL manually

did not work out, unfortunately

Log:
_2022-02-20 19:31:09,430 ERROR[PlexTraktSync]:Plex connection error: HTTPSConnectionPool(host='192-168-178-241.7aec5c30ac80472582085e9f43bd190.plex.direct', port=32400): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x14d890cf6470>: Failed to establish a new connection: [Errno 113] Host is unreachable')), fallback url http://172.17.0.1:32400 didn't respond either._

I'm really thankful for your help, but thats all way to complicated and timeconsuming to me. So I give up.

**Reply by simonc56** [reply, marked as answer]
What you need to do is :
- check in Plex settings that Secure Connections is set to `Preferred`
- remove .env file
- run script to login and recreate proper .env file
- edit .env file at line PLEX_FALLBACKURL to be `http://192.168.178.241:32400` (local address without SSL)

**Reply by glensc** [reply]
> I'm sorry, I have no idea what markdown code fence is and how to use it, but I edited my post anyways.

how about google markdown syntax?

**Reply by glensc** [reply]
also, unclear what you edit, your ` .env` is still without code block, so can't see what the protocol is in the url fields
